src/spikeinterface/sortingcomponents/peak_detection.py

```diff
@@ -63,121 +65,76 @@ def detect_peaks(recording, method='by_channel', pipeline_nodes=None,
     -------
     peaks: array
         Detected peaks.
-
     Notes
     -----
     This peak detection ported from tridesclous into spikeinterface.
     """
 
+
     assert method in detect_peak_methods
 
     method_class = detect_peak_methods[method]
     
     method_kwargs, job_kwargs = split_job_kwargs(kwargs)
-    mp_context = method_class.preferred_mp_context
+    job_kwargs['mp_context'] = method_class.preferred_mp_context
 
-    # prepare args
-    method_args = method_class.check_params(recording, **method_kwargs)
+    node0 = method_class(recording, **method_kwargs)
+    nodes = [node0]
 
-    extra_margin = 0
+    job_name = f'detect peaks using {method}'
     if pipeline_nodes is None:
         squeeze_output = True
     else:
-        check_graph(pipeline_nodes)
-        extra_margin = max(node.get_trace_margin() for node in pipeline_nodes)
         squeeze_output = False
+        job_name += f'  + {len(pipeline_nodes)} nodes'
+
+        # because node are modified inplace (insert parent) they need to copy incase
```

Comment:
What kind of error this could lead to? (modifying in place) Do we have a tests for that?

Reply:
If you run two times the same piepline then this new nodes is add many times.

Reply:
Thanks!